### PR TITLE
feat: support loading public hale connect projects headless

### PR DESCRIPTION
**Pull request by stempler**
Added a new project reader that can be used with a hale connect URN as
source.
Use this reader when the hc protocol is found for a transformation
project in the transformation CLI task.
Currently restricted to public projects (no login performed).

### CODE REVIEW

**Comment by stempler** [comment]
@florianesser The Travis task seems to have failed because of GitLab not being available - is there an easy way to retrigger it?

**Comment by florianesser** [comment]
When logged in, you should see a `Restart build` button on [this page](https://travis-ci.org/halestudio/hale/builds/552710468).

grafik

File: io/plugins/eu.esdihumboldt.hale.io.haleconnect/src/eu/esdihumboldt/hale/io/haleconnect/project/SimpleProjectReader.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		HaleConnectService hcs = getServiceProvider().getService(HaleConnectService.class);
+
+		// TODO login if applicable? (based on reader parameters)
+		// XXX for now only support public projects
+
+		LocatableInputSupplier<? extends InputStream> hcSource;
+		try {
+			hcSource = hcs.loadProject(owner, projectId);
+		} catch (HaleConnectException e) {
+			throw new IllegalStateException();
```
Comment (florianesser) [inline io/plugins/eu.esdihumboldt.hale.io.haleconnect/src/eu/esdihumboldt/hale/io/haleconnect/project/SimpleProjectReader.java]: Should we add the cause here?
```suggestion
			throw new IllegalStateException(e.getMessage(), e);
```

**Comment by stempler** [comment]
retest this please

**Pull request merged**